content/blog/2018-02-07-update-on-async-rendering.md

```diff
@@ -0,0 +1,123 @@
+4. For UI elements like hidden popups and tabs, we'd like to be able to start pre-rendering their content when the browser isn't busy. This way, they can appear instantaneously in response to a later user interaction. However, we don't want to make the initial rendering slower, so it's essential to render such elements lazily ([when the browser is idle](https://developers.google.com/web/updates/2015/08/using-requestidlecallback)).
+5. For many apps, React is not the only JavaScript on the page. It often has to coordinate with other JS libraries, server-rendered widgets, and so on. Asynchronous rendering lets React better coordinate with non-React code regarding when components are inserted into the DOM so that [the user experience is smooth](https://twitter.com/acdlite/status/909926793536094209).
+
+Of course, it's possible to implement some of these features today, but it's difficult. We hope to make them effortless by building them into React itself. By replacing problematic lifecycles with safer alternatives, we also hope to make it simple to write async-safe React components.
+
+In the next section, we'll look at how to update your existing components to prepare for the upcoming lifecycle changes.
+
+## Updating class components
+
+#### If you're an application developer, **you don't have to do anything about the deprecated methods yet**. The primary purpose of this update (v16.3) is to enable open source project maintainers to update their libraries in advance of any deprecation warnings. Those warnings will be enabled with the next minor release, v16.4.
+
+However, if you'd like to start using the new component API (or if you're a maintainer looking to update your library in advance) here are a few examples that we hope will help you to start thinking about components a bit differently. Over time, we plan to add additional “recipes” to our documentation that show how to perform common tasks in a way that's async-safe.
+
+### Initializing state
+
+This example shows a component with `setState` calls inside of `componentWillMount`:
+`embed:update-on-async-rendering/initializing-state-before.js`
+
+The simplest refactor for this type of component is to move the state-updates to the constructor or to a property initializer, like so:
+`embed:update-on-async-rendering/initializing-state-after.js`
+
+### Fetching external data
+
+Here is an example of a component that uses `componentWillMount` to fetch external data::
+`embed:update-on-async-rendering/fetching-external-data-before.js`
+
+The above code is problematic for both server rendering (where the external data won't be used) and the upcoming async rendering mode (where the request might be initiated multiple times, or executed unnecessarily).
+
+The upgrade path for this is to move data-fetching into `componentDidMount`:
+`embed:update-on-async-rendering/fetching-external-data-after.js`
+
```

Comment:
I'd add:

>There is a common misconception that fetching in `componentWillMount` lets you avoid the first empty rendering state. In practice this was never true because React has always executed `render` immediately after `componentWillMount`. If the data is not available by the time `componentWillMount` fires, the first render will still show a loading state regardless of where you initiate the fetch. This is why moving it `componentDidMount` has no negative effect in the vast majority of cases.